source/Octopus.Versioning/Maven/MavenPackageID.cs

```diff
@@ -47,13 +50,12 @@ public MavenPackageID([NotNull]
             if (string.IsNullOrWhiteSpace(version))
                 throw new ArgumentException("Version can not be empty");
 
-            Group = group.Trim();
-            Artifact = artifact.Trim();
-            Version = version.Trim();
+            Group = group!.Trim();
```

Comment:
In .NET 5+ the ` if (string.IsNullOrWhiteSpace(group)) throw new ArgumentException(...);` is sufficient to tell the compiler that group can't be null, however because we're targeting `netstandard2.0` it's not that smart.

I added the `!` simply to suppress the IDE warning